### ISSUE TITLE
Read the final-norm scale from the module instead of a model-family table

### ISSUE THREAD

**Issue by speediedan** [speediedan/interpretune#523]
## Provenance

The approach was established on 2026-09-07 in the norm-folding investigation, not by this issue. This
issue tracks adopting it in-tree, re-measures it with the class list enumerated rather than counted, and
carries three constraints from that record that a naive adoption would drop. The constraints come first,
because each one turns a clean idea into a wrong one if skipped.

## Summary

`optools.py` decides the final norm's elementwise scale from a hardcoded family table:
`_RMSNORM_OFFSET_MODEL_TYPES` (the `(1 + weight)` convention) against
`_RMSNORM_NO_OFFSET_MODEL_TYPES` (plain `weight`), keyed on `config.model_type`.

For any RMSNorm, `norm(x) = (x / rms(x)) * s` and `rms(c * ones) = c`, so a constant vector normalizes
to ones and the module returns its own effective scale. No table, no family prefix:

```python
with torch.no_grad():
    s = norm_module(torch.full((d,), 10.0))   # NOT divided by c: rms() already removed it
```

## The three constraints

1. **Detach.** The probe is a forward pass through a module whose weight is a `Parameter`, so it carries
   `requires_grad=True` (measured: `True` unguarded). Lens directions are consumed as constant bases, so
   an undetached probe attaches the norm's parameters to every graph a caller builds. The seam must
   detach and **assert** it: the point is removing a silent wrong answer, not trading it for a quieter one.
2. **RMSNorm only.** For LayerNorm a constant vector centers to zero and the module returns its **bias**
   (measured: `max|out - bias| = 0.00e+00`). The declared path survives on that arm.
3. **It replaces the scale, not the kind.** The construction needs the scale *and* the norm kind, and the
   kind selects whether centering applies. A probe cannot say whether it is applicable, so a classifier is
   still required ahead of it. Today that classifier is the norm's **class name** rather than a family
   prefix, which is why it was not implicated in the earlier defect.

4. **Do NOT guard the probe with `isinstance(norm, torch.nn.Module)`.** The seam's main consumer is the
   NNsight adapter, where the norm arrives as an `nnsight` `Envoy`, and **`Envoy` is not an `nn.Module`
   subclass** (measured). Such a guard would therefore fall back to the family table on exactly the path
   the seam exists to serve, silently, while looking correct on the HF path where it is easy to test.

   The probe **does** work through the proxy: `Envoy.__call__` delegates to the wrapped module, so
   calling it computes. Measured on `Gemma2RMSNorm` wrapped in an `Envoy`: recovered scale exact against
   `1 + w`, identical to the direct-module result. Guard on the call succeeding and the result being a
   finite tensor shaped like the weight, not on the object's type.

## Measurements

Ten RMSNorm classes, `d = 64`, float32, recovered scale against ground truth. `Gemma3nRMSNorm` and
`Gemma4RMSNorm` are the two the family-prefix rule gets wrong, and the probe handles them with no
special case:

| class | convention | err `c=1` | err `c=10` |
| --- | --- | --- | --- |
| `GemmaRMSNorm`, `Gemma2RMSNorm`, `Gemma3RMSNorm` | `1 + w` | `9.54e-07` | `0.00e+00` |
| `Gemma3nRMSNorm`, `Gemma4RMSNorm` | `w` | `4.77e-07` | `0.00e+00` |
| `LlamaRMSNorm`, `Qwen2RMSNorm`, `MistralRMSNorm`, `Olmo2RMSNorm`, `Phi3RMSNorm` | `w` | `4.77e-07` | `0.00e+00` |

**Use `c >= 10`.** At `c = 1` the `eps` term is not negligible relative to `rms`, giving the ~`1e-6`
error above; by `c = 10` it is exact. (This reproduces the earlier record's `1.4e-6` figure, which was
taken at a different weight scale.)

**Positive control**, because every recovery at `c = 10` is exact and a comparison that cannot fail is
not evidence: checked against the **wrong** convention the same probe reports a max error of exactly
`1.0000` in both directions, which is the offset itself. The check discriminates.

## Why the table is worth removing rather than extending

- **It is a standing liability.** `gemma3n` and the `gemma4` line use plain `weight` while
  `gemma`/`gemma2`/`gemma3` use `(1 + weight)`, so the split runs *inside* one family prefix.
- **A wrong entry is silent**, and scales every direction built from that model. This repository has
  already paid for exactly that once.
- **An unrecognized family has no good answer.** The current code warns and guesses.

## Scope

One forward pass on a `d`-length vector at resolution time, under `no_grad`. It replaces the table, not
the folding: a *direction* still needs the linearized scale, so we cannot simply call the real norm the
way a readout-only implementation can.

**Comment by speediedan** [comment]
## This change must update `docs/jlens_norm_folding.md` in the same PR

Recorded before #473 merges, so the obligation exists on the issue rather than in anyone's head.

#473 lands a page whose "Per-family conventions" section describes the **current** mechanism, correctly
for the tree as it stands:

> `gemma`, `gemma2` and `gemma3` RMSNorms apply `(1 + weight)`; `gemma3n` and the whole `gemma4` line
> apply `weight` directly [...] `gemma3` carries the offset and `gemma3n` does not, so no name prefix
> separates them. [...] The seam therefore selects on **exact family membership** and **warns** when it
> meets a family in that namespace it does not recognize.

That is accurate today and **this issue falsifies its last sentence.** Once the scale is derived from
the module, the seam selects on nothing and warns about nothing: there is no membership test to be
exact about and no unrecognized family to warn on, because an unrecognized family is exactly the case
the probe handles without special treatment.

**What stays true and should not be deleted with it:** the split itself is real and worth keeping as
the reason the mechanism changed. `gemma3` and `gemma3n` genuinely differ, the difference genuinely
does not follow the name, and getting it wrong is genuinely silent. That paragraph is the motivation
for deriving rather than declaring; only the sentence describing *how the seam decides* is superseded.

**Why this is on the issue rather than left to review.** A page that becomes wrong on someone else's
merge is a held-artifact failure, and this page has already had one: an earlier version stated the
`gemma`-by-prefix rule that was correct when written and silently wrong for `gemma3n` and the whole
`gemma4` line. The same page, the same paragraph, the same failure mode one revision apart. A note in
the PR that supersedes it is the only thing that catches it, because nothing in either diff will.

Also update `docs/source/usage/jlens_norm_folding.md` if it carries the passage rather than including
the top-level file.

**Comment by speediedan** [comment]
Substance verified landed by #559 (whose merge message named this issue but could not auto-close it — close keywords only act from PR bodies). Evidence off origin/main: no family table remains in optools.py; _rmsnorm_scale probes with _RMSNORM_PROBE_CONSTANT = 10.0 under no_grad with detach plus a requires_grad assert; shape/finite refusal naming the norm type; the kind classifier still ahead of the probe; tests in test_unembed_norm_seam.py cover one-plus-weight vs weight-directly, missing-norm returns-None, the scale-follows-the-module case and positive controls. Closing the loop #559 left open.
